### PR TITLE
Fix computation of effects

### PR DESCRIPTION
**Pull request by pontusmelke**
o The traversal logic for walking the old ast was buggy and untested

### CODE REVIEW

File: community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/mutation/CreateNode.scala [hunk 1]
```diff
@@ -34,6 +34,7 @@ case class CreateNode(key: String, properties: Map[String, Expression], labels:
   with GraphElementPropertyFunctions
   with CollectionSupport {
 
+
```
Comment (davidegrohmann) [inline community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/mutation/CreateNode.scala]: revert this

File: community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/mutation/UpdateAction.scala [hunk 2]
```diff
@@ -48,17 +50,55 @@ trait UpdateAction extends TypeSafe with AstNode[UpdateAction] {
...
   }
 
   // This is here to give FOREACH action a chance to introduce symbols
-  def updateSymbols(symbol: SymbolTable ): SymbolTable = symbol
+  def updateSymbols(symbol: SymbolTable): SymbolTable = symbol.add(identifiers.toMap)
 
   def localEffects(symbols: SymbolTable): Effects
 }
+
+class EffectsCollector(var _effects: Effects, expr: AstNode[_], symbolTable: SymbolTable) {
```
Comment (davidegrohmann) [inline community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/mutation/UpdateAction.scala]: maybe `_effect` should be private?

File: community/cypher/cypher-compiler-2.2/src/test/scala/org/neo4j/cypher/internal/compiler/v2_2/mutation/EffectfulTest.scala [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import org.neo4j.cypher.internal.commons.CypherFunSuite
+import org.neo4j.cypher.internal.compiler.v2_2.ast.convert.commands.StatementConverters._
+import org.neo4j.cypher.internal.compiler.v2_2.commands.Query
+import org.neo4j.cypher.internal.compiler.v2_2.commands.expressions.{Identifier, Literal, Property}
+import org.neo4j.cypher.internal.compiler.v2_2.commands.values.TokenType.PropertyKey
+import org.neo4j.cypher.internal.compiler.v2_2.executionplan.Effects
+import org.neo4j.cypher.internal.compiler.v2_2.symbols
+import org.neo4j.cypher.internal.compiler.v2_2.symbols.SymbolTable
+
+class EffectfulTest extends CypherFunSuite {
```
Comment (davidegrohmann) [inline community/cypher/cypher-compiler-2.2/src/test/scala/org/neo4j/cypher/internal/compiler/v2_2/mutation/EffectfulTest.scala]: This is very nice, we might want to expand it to cover more tricky cases. On the other hand this should not block the PR to be merged and can be done separately.

**Comment by Davidmylove** [comment]
Ya but you won't have erica

**Pull request merged**